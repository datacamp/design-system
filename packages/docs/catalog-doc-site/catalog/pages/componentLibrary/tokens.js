import tokens from '@datacamp/waffles-tokens';
import { Page } from 'catalog';
import React from 'react';

import CustomHeader from '../../components/CustomHeader';

export default () => {
  return (
    <main>
      <CustomHeader
        section="Component Library"
        subSection="Tokens"
        description="The primitive values used to style anything and everything"
      />

      <Page>
        <section className="dc-u-maxw-100pc">
          <div className="dc-card dc-u-p-24 dc-u-mt-8">
            <h4>About</h4>
            <p>
              The <code>@datacamp/waffles-tokens</code> package exposes a json
              file containing all the primitive values necessary to produce apps
              with the datacamp styling rules. These values can be accessed and
              used to style applications using a variety of tools, including
              CSS-in-JS libraries such as emotion.
            </p>
            <h4>Usage</h4>
            <p>
              To use any of the styles, simply import the{' '}
              <code>@datacamp/waffles-tokens</code> package and directly access
              the necessary properties. For example if the primary color is
              needed then it should be accessed as follows.
            </p>
            <pre>
              <code>{`import { colors } from '@datacamp/waffles-tokens';

const primaryColorValue = colors.primary;`}</code>
            </pre>
            <p>
              This hex code can then be used wherever is necessary. All the
              tokens can be accessed and used in the same way since it is just a
              json file.
            </p>
            <h4>Available tokens</h4>
            <pre>
              <code>{JSON.stringify(tokens, null, 2)}</code>
            </pre>
          </div>
        </section>
      </Page>
    </main>
  );
};
