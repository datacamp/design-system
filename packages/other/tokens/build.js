const styleDictionary = require('style-dictionary');

const coreJsonFormatter = require('./formats/coreJson');
const jsonFormatter = require('./formats/jsonFormatter');
const scssVariableFormatter = require('./formats/scssVariableFormatter');
const gradientTransform = require('./transforms/gradient');

const transforms = ['attribute/cti', 'attribute/color', 'waffles/gradient'];

const tokenBuilder = styleDictionary.extend({
  platforms: {
    'core-json': {
      buildPath: 'lib/',
      files: [{ destination: 'core-tokens.json', format: 'waffles/core-json' }],
      transforms,
    },
    'full-json': {
      buildPath: 'lib/',
      files: [{ destination: 'future-tokens.json', format: 'waffles/json' }],
      transforms,
    },
    scss: {
      buildPath: 'lib/',
      files: [
        { destination: 'variables.scss', format: 'waffles/scss-variables' },
      ],
      transforms,
    },
  },
  source: ['properties/**/*.json'],
});

tokenBuilder.registerTransform({
  matcher: (prop) => prop.attributes.category === 'gradient',
  name: 'waffles/gradient',
  transformer: gradientTransform,
  type: 'value',
});

tokenBuilder.registerFormat({
  formatter: coreJsonFormatter,
  name: 'waffles/core-json',
});

tokenBuilder.registerFormat({
  formatter: jsonFormatter,
  name: 'waffles/json',
});

tokenBuilder.registerFormat({
  formatter: scssVariableFormatter,
  name: 'waffles/scss-variables',
});

tokenBuilder.buildAllPlatforms();
