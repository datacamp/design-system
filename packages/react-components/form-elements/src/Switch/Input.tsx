import { css } from '@emotion/react';
import { forwardRef, InputHTMLAttributes } from 'react';

// Checkbox is visually hidden but remains accessible to screen readers
const inputStyle = css({
  border: 0,
  height: '100%',
  left: 0,
  margin: 0,
  opacity: 0.001,
  padding: 0,
  position: 'absolute',
  top: 0,
  width: '100%',
  zIndex: -1,
});

const Input = forwardRef<
  HTMLInputElement,
  InputHTMLAttributes<HTMLInputElement>
>((props, ref) => {
  return <input css={inputStyle} ref={ref} {...props} />;
});

export default Input;
