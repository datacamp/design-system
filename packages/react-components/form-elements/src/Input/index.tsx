import { HiddenIcon, VisibleIcon } from '@datacamp/waffles-icons';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { computeDataAttributes } from '@datacamp/waffles-utils';
import { ClassNames, css } from '@emotion/react';
import React, { forwardRef, ReactElement, Ref } from 'react';

import {
  fontSizes,
  heights,
  iconSize,
  inputPaddings,
  inputStyle,
  inputWithIconPaddings,
} from '../formStyles';
import Label from '../Label';

interface InputProps {
  /**
   * Sets the autocomplete attribute on the rendered input element.
   */
  autocomplete?: string;
  /**
   * Sets the css className on the rendered element. Can be used to add custom
   * styling.
   */
  className?: string;
  /**
   * An object who's keys will be prepended with 'data-', and values will be
   * used as attributes on the rendered element.
   */
  dataAttributes?: { [key: string]: string };
  /**
   * When provided the description will show underneath the label. Only
   * available when 'label' is also specified.
   */
  description?: string;
  /**
   * Disables the input when true.
   */
  disabled?: boolean;
  /**
   * When provided the errorMessage will show underneath the input. Only
   * available when label is also specified.
   */
  errorMessage?: string;
  /**
   * Sets the html attribute "required". Has no visual impact. Use the
   * "required" prop to set visuals if required.
   */
  htmlRequired?: boolean;
  /**
   * When an icon is provided it will render inside the input. This can only be
   * an component from the @datacamp/waffles-icons package.
   */
  icon?: ReactElement;
  /**
   * Sets the html id on the rendered input element.
   */
  id?: string;
  /**
   * Renders label text above the input. If this is not provided the input will
   * render inline. When a label is provided many other props are also available
   * and this component becomes a block that can be stacked with other form
   * elements.
   */
  label?: string;
  /**
   * The maximum value of the input.
   */
  max?: string | number;
  /**
   * Sets the maxLength attribute on the rendered input.
   */
  maxLength?: number;
  /**
   * The minimum value of the input.
   */
  min?: string | number;
  /**
   * Used to set the html name attribute. Uniquely indentifies the input within
   * the current form context.
   */
  name: string;
  /**
   * Called when the user removes focus from the input.
   */
  onBlur?: () => void;
  /**
   * Called when the user requests a change to the value of the input. This
   * should be used to set the value.
   */
  onChange: (value: string) => void;
  /**
   * Called when the user focuses the input
   */
  onFocus?: () => void;
  /**
   * The placeholder text to render before the user has entered a value.
   */
  placeholder?: string;
  /**
   * It defines wheter the input field is required or not. If required=true it
   * adds the text 'Required' on the top-right of the input, if required=false
   * it adds 'Optional'. The default value is undefined, which doesn't add
   * anything. This is only available when 'label' is also provided.
   */
  required?: boolean;
  /**
   * The size of the input element.
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * The step size for a number input.
   */
  step?: string | number;
  /**
   * The type of input to render. This corresponds to a set of html input types.
   */
  type?:
    | 'text'
    | 'date'
    | 'datetime-local'
    | 'email'
    | 'month'
    | 'number'
    | 'password'
    | 'search'
    | 'tel'
    | 'time'
    | 'url'
    | 'week';

  /**
   * The current value of the input. This should be controlled by listening to
   * onChange.
   */
  value: string;
}

const InternalInput = ({
  autocomplete,
  className,
  dataAttributes,
  description,
  disabled = false,
  errorMessage,
  htmlRequired,
  icon,
  id,
  innerRef,
  label,
  max,
  maxLength,
  min,
  name,
  onBlur,
  onChange,
  onFocus,
  placeholder,
  required,
  size = 'medium',
  step,
  type = 'text',
  value,
}: InputProps & { innerRef?: Ref<HTMLInputElement> }): ReactElement => {
  const [passwordVisible, setPasswordVisible] = React.useState(false);
  const togglePasswordVisibility = React.useCallback(
    () => setPasswordVisible(!passwordVisible),
    [passwordVisible],
  );

  const parsedDataAttributes = computeDataAttributes(dataAttributes);

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void =>
    onChange(event.target.value);

  const handleBlur = (): void => onBlur && onBlur();
  const handleFocus = (): void => onFocus && onFocus();

  const inputSize = css(
    { fontSize: fontSizes[size], height: heights[size] },
    inputPaddings[size],
  );

  const getInputStyle = css(
    inputSize,
    inputStyle,
    icon && inputWithIconPaddings[size],
    label && css({ width: '100%' }),
  );

  const iconStyle = css({
    display: 'inline-block',
    left: tokens.size.space[12].value,
    position: 'absolute',
    top: (heights[size] - iconSize[size]) / 2,
  });

  const iconElement = icon && (
    <div css={{ position: 'relative' }}>
      <ClassNames>
        {({ css: getClassName }) =>
          React.cloneElement(icon, {
            className: getClassName(iconStyle),
            size: 'medium',
          })
        }
      </ClassNames>
    </div>
  );

  const showHidePasswordElement = type === 'password' && (
    <button
      aria-label={`${passwordVisible ? 'Hide' : 'Show'} Password`}
      css={{
        '&:active, &:focus, &:hover': {
          opacity: 1,
          outline: 0,
        },
        background: 'transparent',
        border: 0,
        color: tokens.color.primary.navyText.value.hex,
        display: 'block',
        opacity: 0.5,
        position: 'absolute',
        right: 12,
        top: 11,
      }}
      onClick={togglePasswordVisibility}
      type="button"
    >
      <i aria-hidden="true">
        {passwordVisible ? (
          <HiddenIcon size="medium" />
        ) : (
          <VisibleIcon size="medium" />
        )}
      </i>
    </button>
  );

  const inputElement = (
    <div
      css={{ display: label ? 'block' : 'inline-block', position: 'relative' }}
    >
      {iconElement}
      <input
        autoComplete={autocomplete}
        className={className}
        css={getInputStyle}
        disabled={disabled}
        id={id}
        max={max}
        maxLength={maxLength}
        min={min}
        name={name}
        onBlur={handleBlur}
        onChange={handleChange}
        onFocus={handleFocus}
        placeholder={placeholder}
        ref={innerRef}
        required={htmlRequired}
        step={step}
        type={type === 'password' && passwordVisible ? 'input' : type}
        value={value}
        {...parsedDataAttributes}
      />
      {showHidePasswordElement}
    </div>
  );

  return (
    <>
      {label ? (
        <Label
          description={description}
          errorMessage={errorMessage}
          htmlFor={id}
          label={label}
          required={required}
        >
          {inputElement}
        </Label>
      ) : (
        <>{inputElement}</>
      )}
    </>
  );
};

const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => (
  <InternalInput innerRef={ref} {...props} />
));

export default Input;
