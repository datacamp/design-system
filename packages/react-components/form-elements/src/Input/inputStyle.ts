import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';

const inputStyle = css({
  '::placeholder': {
    color: tokens.color.opaque.greyOslo.value.rgb,
    fontFamily: 'inherit',
  },

  ':disabled, :active:disabled, :focus:disabled': {
    cursor: 'not-allowed',
  },
  ':disabled::placeholder': { color: '#D1D3D8' },

  ':focus': {
    boxShadow: `inset 0 0 0 1px ${tokens.color.opaque.primary.value.rgb}`,
    outline: 'none',
  },
  background: 'white',
  border: 0,
  borderRadius: tokens.radii.small.value,
  boxShadow: `inset 0 0 0 1px ${tokens.color.opaque.greyLight.value.rgb}`,
  boxSizing: 'border-box',
  color: tokens.color.opaque.grey.value.rgb,
  display: 'inline-block',
  fontFamily: [
    tokens.asset.font.sansSerif.attributes.fallback,
    tokens.asset.font.sansSerif.value,
  ],
  margin: 0,
  verticalAlign: 'baseline',
});

const inputSizes = {
  large: {
    fontSize: 20,
    height: tokens.size.space[64].value,
    padding: `0 ${tokens.size.space[24].value}px  `,
  },
  medium: {
    fontSize: 16,
    height: tokens.size.space[48].value,
    padding: `0 ${tokens.size.space[16].value}px  `,
  },
  small: {
    fontSize: 16,
    height: tokens.size.space[36].value,
    padding: `0 ${tokens.size.space[12].value}px  `,
  },
};

const paddings = {
  large: { padding: `0 ${tokens.size.space[48].value}px` },
  medium: { padding: `0 ${tokens.size.space[36].value}px  ` },
  small: { padding: `0 ${tokens.size.space[36].value}px  ` },
};

const iconSize = {
  large: 24,
  medium: 18,
  small: 12,
};

export { inputSizes, inputStyle, iconSize, paddings };
