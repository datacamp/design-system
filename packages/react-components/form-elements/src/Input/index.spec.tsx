import '@testing-library/jest-dom/extend-expect';

import { AddCircleIcon } from '@datacamp/waffles-icons';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import Input from '.';

describe('<Input />', () => {
  describe('without label', () => {
    // cannot use axeRender here as no label will fail

    const testName = 'test-name';
    const testValue = 'test value';

    it('sets the correct name and value on the input', () => {
      const { getByRole } = render(
        <Input name={testName} onChange={() => {}} value={testValue} />
      );

      const inputElement = getByRole('textbox') as HTMLElement;

      expect(inputElement).toHaveAttribute('name', testName);
      expect(inputElement).toHaveAttribute('value', testValue);
    });

    it('sets the autocomplete attribute', () => {
      const testAutocomplete = 'off';
      const { getByRole } = render(
        <Input
          autocomplete={testAutocomplete}
          name={testName}
          onChange={() => {}}
          value={testValue}
        />
      );

      const inputElement = getByRole('textbox') as HTMLElement;

      expect(inputElement).toHaveAttribute('autocomplete', testAutocomplete);
    });

    it('sets the className ', () => {
      const testClassName = 'test-class';
      const { getByRole } = render(
        <Input
          className={testClassName}
          name={testName}
          onChange={() => {}}
          value={testValue}
        />
      );
      const inputElement = getByRole('textbox') as HTMLElement;

      expect(inputElement).toHaveClass(testClassName);
    });

    it('sets the data attributes', () => {
      const testDataAttributes = { test: 'example' };
      const { getByRole } = render(
        <Input
          dataAttributes={testDataAttributes}
          name={testName}
          onChange={() => {}}
          value={testValue}
        />
      );
      const inputElement = getByRole('textbox') as HTMLElement;

      expect(inputElement).toHaveAttribute('data-test', 'example');
    });

    it('sets the maxLength', () => {
      const maxLength = 5;
      const { getByRole } = render(
        <Input
          maxLength={maxLength}
          name={testName}
          onChange={() => {}}
          value={testValue}
        />
      );

      const inputElement = getByRole('textbox') as HTMLElement;

      expect(inputElement).toHaveAttribute('maxLength', maxLength.toString());
    });

    it('calls the onBlur function', () => {
      const onBlur = jest.fn();
      const { getByRole } = render(
        <Input
          name={testName}
          onBlur={onBlur}
          onChange={() => {}}
          value={testValue}
        />
      );

      const inputElement = getByRole('textbox') as HTMLElement;

      fireEvent.blur(inputElement);
      expect(onBlur).toHaveBeenCalledTimes(1);
    });

    it('calls the onChange function', () => {
      const onChange = jest.fn();
      const { getByRole } = render(
        <Input name={testName} onChange={onChange} value={testValue} />
      );
      const testString = 'test string';
      const inputElement = getByRole('textbox') as HTMLElement;

      userEvent.type(inputElement, testString);
      expect(onChange).toHaveBeenCalledWith(testString);
    });

    it('sets the placeholder', () => {
      const placeholder = 'placeholder text';
      const { getByRole } = render(
        <Input
          name={testName}
          onChange={() => {}}
          placeholder={placeholder}
          value={testValue}
        />
      );
      expect(getByRole('textbox') as HTMLElement).toHaveAttribute(
        'placeholder',
        placeholder
      );
    });

    it('disables the input if disabled is passed as a prop', () => {
      const { getByRole } = render(
        <Input name={testName} onChange={() => {}} value={testValue} disabled />
      );
      expect(getByRole('textbox')).toHaveAttribute('disabled');
    });

    it('renders an icon inside the input field', async () => {
      const placeholder = 'placeholder text';

      const { container, getByTitle } = await render(
        <Input
          icon={<AddCircleIcon />}
          name={testName}
          onChange={() => {}}
          placeholder={placeholder}
          value={testValue}
        />
      );

      const iconElement = getByTitle('Add') as HTMLElement;
      expect(container.firstChild).toContainElement(iconElement);
    });

    it('sets the id', () => {
      const testId = 'name input';
      const { getByRole } = render(
        <Input
          id={testId}
          name={testName}
          onChange={() => {}}
          value={testValue}
        />
      );
      expect(getByRole('textbox')).toHaveAttribute('id', testId);
    });

    it('disables the input if "disabled" is passed as a prop', () => {
      const placeholder = 'placeholder text';
      const { getByRole } = render(
        <Input
          name={testName}
          onChange={() => {}}
          placeholder={placeholder}
          value={testValue}
          disabled
        />
      );
      expect(getByRole('textbox')).toHaveAttribute('disabled');
    });

    it('renders a small text input if size= "small" ', () => {
      const placeholder = 'placeholder text';
      const { container } = render(
        <Input
          name={testName}
          onChange={() => {}}
          placeholder={placeholder}
          size="small"
          value={testValue}
        />
      );
      expect(container.firstChild).toHaveStyle(`height: 36px;`);
      expect(container.firstChild).toHaveStyle(`font-size: 16px;`);
    });

    it('renders a medium text input if size= "medium" or undefined', () => {
      const placeholder = 'placeholder text';
      const { container } = render(
        <Input
          name={testName}
          onChange={() => {}}
          placeholder={placeholder}
          value={testValue}
        />
      );
      expect(container.firstChild).toHaveStyle(`height: 48px;`);
      expect(container.firstChild).toHaveStyle(`font-size: 16px;`);
    });

    it('renders a medium text input if size= "large" ', () => {
      const placeholder = 'placeholder text';
      const { container } = render(
        <Input
          name={testName}
          onChange={() => {}}
          placeholder={placeholder}
          size="large"
          value={testValue}
        />
      );
      expect(container.firstChild).toHaveStyle(`height: 64px;`);
      expect(container.firstChild).toHaveStyle(`font-size: 20px;`);
    });

    it('adds an error message under the input field if error is passed as a prop', () => {
      const testLabel = 'label text';
      const testError = 'Test error';

      const { getByText } = render(
        <Input
          errorMessage={testError}
          label={testLabel}
          name={testName}
          onChange={() => {}}
          value={testValue}
        />
      );
      expect(getByText(testError)).toBeInTheDocument();
      expect(getByText(testError)).toHaveStyle(
        `color: ${tokens.color.opaque.red.value.hex}`
      );
      expect(getByText(testError)).toHaveStyle(`font-size: 14px`);
    });

    it('renders the indication "required" above the input field if required is passed as a prop ', () => {
      const testLabel = 'label text';

      const { getByText } = render(
        <Input
          label={testLabel}
          name={testName}
          onChange={() => {}}
          value={testValue}
          required
        />
      );
      expect(getByText('Required')).toBeInTheDocument();
      expect(getByText('Required')).toHaveStyle(
        `color: ${tokens.color.opaque.greyOslo.value.hex};`
      );
      expect(getByText('Required')).toHaveStyle(`font-size: 14px;`);
    });

    it('renders the indication "Optional" above the input field if required={false} is passed as a prop ', () => {
      const testLabel = 'label text';

      const { getByText } = render(
        <Input
          label={testLabel}
          name={testName}
          onChange={() => {}}
          required={false}
          value={testValue}
        />
      );
      expect(getByText('Optional')).toBeInTheDocument();
      expect(getByText('Optional')).toHaveStyle(
        `color: ${tokens.color.opaque.greyOslo.value.hex};`
      );
      expect(getByText('Optional')).toHaveStyle(`font-size: 14px;`);
    });
  });
});

describe('with label', () => {
  const testName = 'test name';
  const testValue = 'test value';
  it('sets the label', () => {
    const testLabel = 'label text';
    const { getByLabelText } = render(
      <Input
        label={testLabel}
        name={testName}
        onChange={() => {}}
        value={testValue}
      />
    );
    expect(getByLabelText(testLabel)).toBeInTheDocument();
  });

  it('sets the description', () => {
    const testLabel = 'label text';
    const testDescription = 'description text';
    const { getByText } = render(
      <Input
        description={testDescription}
        label={testLabel}
        name={testName}
        onChange={() => {}}
        value={testValue}
      />
    );
    expect(getByText(testDescription)).toBeInTheDocument();
  });
});

describe('snapshots', () => {
  const exampleText = 'input text';

  const inputSizes: ('small' | 'medium' | 'large')[] = [
    'small',
    'medium',
    'large',
  ];
  const disabled: boolean[] = [true, false];
  const required: (boolean | undefined)[] = [undefined, true, false];

  inputSizes.forEach(size => {
    required.forEach(isRequired => {
      disabled.forEach(isDisabled => {
        it(`renders an input with size ${size}, disabled=${isDisabled}, required=${isRequired}`, async () => {
          const { container } = await render(
            <Input
              disabled={isDisabled}
              errorMessage="test error"
              label="test label"
              name={exampleText}
              onChange={() => {}}
              placeholder={exampleText}
              required={isRequired}
              size={size}
              value=""
            />
          );
          expect(container.firstChild).toMatchSnapshot();
        });
      });
    });
  });
});
