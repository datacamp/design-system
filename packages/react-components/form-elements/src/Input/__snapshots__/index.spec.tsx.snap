// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders an input with size large and disabled false 1`] = `
.emotion-0 {
  font-size: 20px;
  height: 64px;
  padding: 0 24;
  background: white;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgb(209,211,216);
  box-sizing: border-box;
  color: rgb(77,83,86);
  display: inline-block;
  font-family: sans-serif;
  font-family: Lato;
  margin: 0;
  padding: 8px 16px;
  vertical-align: baseline;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::-moz-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled {
  cursor: not-allowed;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::-moz-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::placeholder {
  color: #D1D3D8;
}

.emotion-0:focus {
  box-shadow: inset 0 0 0 1px rgb(51,170,204);
  outline: none;
}

<input
  class="emotion-0"
  id="input text"
  name="input text"
  placeholder="input text"
  value=""
/>
`;

exports[`snapshots renders an input with size large and disabled true 1`] = `
.emotion-0 {
  font-size: 20px;
  height: 64px;
  padding: 0 24;
  background: white;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgb(209,211,216);
  box-sizing: border-box;
  color: rgb(77,83,86);
  display: inline-block;
  font-family: sans-serif;
  font-family: Lato;
  margin: 0;
  padding: 8px 16px;
  vertical-align: baseline;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::-moz-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled {
  cursor: not-allowed;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::-moz-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::placeholder {
  color: #D1D3D8;
}

.emotion-0:focus {
  box-shadow: inset 0 0 0 1px rgb(51,170,204);
  outline: none;
}

<input
  class="emotion-0"
  disabled=""
  id="input text"
  name="input text"
  placeholder="input text"
  value=""
/>
`;

exports[`snapshots renders an input with size medium and disabled false 1`] = `
.emotion-0 {
  font-size: 16px;
  height: 48px;
  padding: 0 16;
  background: white;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgb(209,211,216);
  box-sizing: border-box;
  color: rgb(77,83,86);
  display: inline-block;
  font-family: sans-serif;
  font-family: Lato;
  margin: 0;
  padding: 8px 16px;
  vertical-align: baseline;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::-moz-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled {
  cursor: not-allowed;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::-moz-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::placeholder {
  color: #D1D3D8;
}

.emotion-0:focus {
  box-shadow: inset 0 0 0 1px rgb(51,170,204);
  outline: none;
}

<input
  class="emotion-0"
  id="input text"
  name="input text"
  placeholder="input text"
  value=""
/>
`;

exports[`snapshots renders an input with size medium and disabled true 1`] = `
.emotion-0 {
  font-size: 16px;
  height: 48px;
  padding: 0 16;
  background: white;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgb(209,211,216);
  box-sizing: border-box;
  color: rgb(77,83,86);
  display: inline-block;
  font-family: sans-serif;
  font-family: Lato;
  margin: 0;
  padding: 8px 16px;
  vertical-align: baseline;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::-moz-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled {
  cursor: not-allowed;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::-moz-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::placeholder {
  color: #D1D3D8;
}

.emotion-0:focus {
  box-shadow: inset 0 0 0 1px rgb(51,170,204);
  outline: none;
}

<input
  class="emotion-0"
  disabled=""
  id="input text"
  name="input text"
  placeholder="input text"
  value=""
/>
`;

exports[`snapshots renders an input with size small and disabled false 1`] = `
.emotion-0 {
  font-size: 16px;
  height: 36px;
  padding: 0 12;
  background: white;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgb(209,211,216);
  box-sizing: border-box;
  color: rgb(77,83,86);
  display: inline-block;
  font-family: sans-serif;
  font-family: Lato;
  margin: 0;
  padding: 8px 16px;
  vertical-align: baseline;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::-moz-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled {
  cursor: not-allowed;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::-moz-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::placeholder {
  color: #D1D3D8;
}

.emotion-0:focus {
  box-shadow: inset 0 0 0 1px rgb(51,170,204);
  outline: none;
}

<input
  class="emotion-0"
  id="input text"
  name="input text"
  placeholder="input text"
  value=""
/>
`;

exports[`snapshots renders an input with size small and disabled true 1`] = `
.emotion-0 {
  font-size: 16px;
  height: 36px;
  padding: 0 12;
  background: white;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgb(209,211,216);
  box-sizing: border-box;
  color: rgb(77,83,86);
  display: inline-block;
  font-family: sans-serif;
  font-family: Lato;
  margin: 0;
  padding: 8px 16px;
  vertical-align: baseline;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::-moz-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:-ms-input-placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0::placeholder {
  color: rgb(133,144,148);
  font-family: inherit;
}

.emotion-0:disabled,
.emotion-0:active:disabled,
.emotion-0:focus:disabled {
  cursor: not-allowed;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::-moz-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #D1D3D8;
}

.emotion-0:disabled::placeholder {
  color: #D1D3D8;
}

.emotion-0:focus {
  box-shadow: inset 0 0 0 1px rgb(51,170,204);
  outline: none;
}

<input
  class="emotion-0"
  disabled=""
  id="input text"
  name="input text"
  placeholder="input text"
  value=""
/>
`;
